### PR TITLE
Add the mean reduction for mse_loss

### PR DESCRIPTION
**Pull request by libratiger**
## Description
Add the mean reduction for mse_loss solves issue https://github.com/OpenMined/SyMPC/issues/176

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
complete the already exist test file

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by aanurraj — COMMENTED** [review]
Also run pre-commit hook.

File: tests/sympc/module/nn/functional_test.py [hunk 1]
```diff
@@ -34,7 +34,16 @@ def test_mse_loss(get_clients) -> None:
     y_pred = torch.Tensor([0.1, 0.3, 0.4, 0.2])
     y_pred_mpc = MPCTensor(secret=y_pred, session=session)
```
[the remark this answers is not in the document]
Comment (libratiger) [reply tests/sympc/module/nn/functional_test.py:35]: I will add later by a new commit
```suggestion
    y_pred_mpc = MPCTensor(secret=y_pred, session=session)
```

[the remark this answers is not in the document]
Comment (libratiger) [reply tests/sympc/module/nn/functional_test.py:37]: I just fix this

**Comment by libratiger** [comment]
> Also run pre-commit hook.

It's Ok now

**Comment by libratiger** [comment]
there is no  `len` method for `torch.tensor`, is there any design consideration. i just open one issue: https://github.com/OpenMined/SyMPC/issues/182

File: src/sympc/module/nn/functional.py [hunk 2]
```diff
@@ -16,14 +16,24 @@ def relu(x: MPCTensor) -> MPCTensor:
...
 
     Returns:
         The loss
+
+    Raises:
+        ValueError: If `reduction` not in supported methods
     """
-    return (pred - target).pow(2).sum()
+    if reduction == "mean":
+        result = (pred - target).pow(2).sum() * 1.0 / len(pred)
```
Comment (gmuraru) [inline src/sympc/module/nn/functional.py:34]: Q: Why multiply by 1.0?
Why not use size rather than len?

Comment (libratiger) [reply src/sympc/module/nn/functional.py:34]: the sum maybe a int, but the mean should be flout

Comment (gmuraru) [reply src/sympc/module/nn/functional.py:34]: in MPC it should still be an integer

Comment (libratiger) [reply src/sympc/module/nn/functional.py:34]: I will fix with your advice, but can you explain to me with more detail?

Comment (gmuraru) [reply src/sympc/module/nn/functional.py:34]: Sure.

We have the `MPCTensor` class which holds tensor to `ShareTensor`s (the secret value is split across multiple parties which holds those `ShareTensors`). `ShareTensor`s are in `FixedPointPrecision` (integers).

When we use this `encoding` with `FixedPointPrecision` we transform numbers like the following:
`123.23232` ---> `12323` (this happens if we use a base of 10 and a precision of 2).

All our values are encoded and they are decoded back when we want to reconstruct the result of the computation.

Comment (libratiger) [reply src/sympc/module/nn/functional.py]: Thanks, I update the fix

**Review by aanurraj — APPROVED** [review]
Looks Good 🎉

**Pull request merged**